### DISCUSSION TITLE
Is possible to configure different DbType for a column for different database (or IDataProvider)

### DISCUSSION THREAD

**Discussion by alexey-leonovich** [linq2db/linq2db#4499] [Q&A] [answered]
Hello.
Depending on config file my app connects to SQL Server Compact or SQLite database. In one of my domain objects I have a field representing transaction sum (decimal C# type). 
In SQL Server Compact I can safely use `[Column(DbType = "numeric(29,4)"]` annotation as this database supports decimal natively. 
SQLite uses REAL affinity (floating-point) for a decimal value and promises to preserve [the 15 most significant digits](https://www.sqlite.org/datatype3.html). So to preserve precision in my computations I want to use `[Column(DbType = "numeric(15,4)"]` annotation, but only for SQLite.
Is possible to configure different DbType for a column for different database (or IDataProvider)? For example in EF Core it can be done [inside](https://github.com/dotnet/efcore/issues/18756#issuecomment-549964817) `protected override void OnModelCreating(ModelBuilder builder)`. Thank you.

**Comment by MaceWindu** [comment, marked as answer]
You can use multiple mapping attributes with different `Configuration`, which should contain connection name (or at least provider name):
```cs
[Column(DataType = DataType.Decimal, Configration = "MySqlCeConnection")]
[Column(DataType = DataType.Double, Configration = "MySQLiteConnection")]

```
